src/braket/aws/aws_quantum_task_batch.py

```diff
@@ -130,24 +130,30 @@ def _execute(
                 )
         max_threads = min(max_parallel, max_workers)
         remaining = [0 for _ in task_specifications]
-        with ThreadPoolExecutor(max_workers=max_threads) as executor:
-            task_futures = [
-                executor.submit(
-                    AwsQuantumTaskBatch._create_task,
-                    remaining,
-                    aws_session,
-                    device_arn,
-                    task,
-                    s3_destination_folder,
-                    shots,
-                    poll_timeout_seconds=poll_timeout_seconds,
-                    poll_interval_seconds=poll_interval_seconds,
-                    *args,
-                    **kwargs,
-                )
-                for task in task_specifications
-            ]
-        tasks = [future.result() for future in task_futures]
+        try:
+            with ThreadPoolExecutor(max_workers=max_threads) as executor:
+                task_futures = [
+                    executor.submit(
+                        AwsQuantumTaskBatch._create_task,
+                        remaining,
+                        aws_session,
+                        device_arn,
+                        task,
+                        s3_destination_folder,
+                        shots,
+                        poll_timeout_seconds=poll_timeout_seconds,
+                        poll_interval_seconds=poll_interval_seconds,
+                        *args,
+                        **kwargs,
+                    )
+                    for task in task_specifications
+                ]
+            tasks = [future.result() for future in task_futures]
+        except:  # noqa: E722
+            # If we are unable to return the tasks, attempt to cancel any pending tasks
+            for future in task_futures:
```

Comment:
Do we want to log that we are canceling all remaining tasks or some sort of logging?

Reply:
I think the initial comment was a bit unclear. I think if tasks were being cancelled, that would be worth noting. I have edited it to clarify that tasks are not being cancelled on abortion. The only action is to stop creating new tasks. Since the exception is then reraised to be handled elsewhere, I do not think logging would add as much value here.